L1Trigger/Phase2L1ParticleFlow/interface/jetmet/L1PFJUMPEmulator.h

```diff
@@ -0,0 +1,282 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_JUMP_h
+#define L1Trigger_Phase2L1ParticleFlow_JUMP_h
+
+#include "DataFormats/L1TParticleFlow/interface/jets.h"
+#include "DataFormats/L1TParticleFlow/interface/sums.h"
+#include "DataFormats/L1TParticleFlow/interface/layer1_emulator.h"
+
+#include "L1Trigger/Phase2L1ParticleFlow/interface/dbgPrintf.h"
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include "ap_int.h"
+#include "ap_fixed.h"
+
+namespace L1JUMPEmu {
+  // Emulator for the JUMP Algorithm
+  // JUMP: Jet Uncertainty-aware MET Prediction
+
+  typedef l1ct::pt_t pt_t;
+  typedef l1ct::glbphi_t phi_t;
+  typedef l1ct::glbeta_t eta_t;
+  typedef l1ct::dpt_t pxy_t;
+  typedef l1ct::Sum Met;
+
+  typedef ap_fixed<22, 12> proj_t;
+  typedef ap_fixed<32, 22> proj2_t;
+  typedef ap_fixed<32, 2> poly_t;
+  typedef ap_fixed<32, 2> poly2_t;
+  // typedef ap_fixed<64, 32> proj_t;
+  // typedef ap_fixed<64, 32> proj2_t;
+
+  struct Particle_xy {
+    // 44bits
+    proj_t hwPx;  // 22bits
+    proj_t hwPy;  // 22bits
+  };
+
+  inline Particle_xy Get_xy(l1ct::Sum in_particle) {
```

Comment:
Define this function or something like it in another file and call it from both `L1PFJUMPEmulator.h` and `L1PFMetEmulator.h` to avoid code duplication